src/command_modules/azure-cli-profile/azure/cli/command_modules/profile/custom.py

```diff
@@ -92,14 +92,16 @@ def account_clear(cmd):
 
 # pylint: disable=inconsistent-return-statements
 def login(cmd, username=None, password=None, service_principal=None, tenant=None, allow_no_subscriptions=False,
-          identity=False):
+          identity=False, use_device_code=False):
     """Log in to access Azure subscriptions"""
     from adal.adal_error import AdalError
     import requests
 
     # quick argument usage check
-    if (any([password, service_principal, tenant, allow_no_subscriptions]) and identity):
+    if any([password, service_principal, tenant, allow_no_subscriptions]) and identity:
         raise CLIError("usage error: '--identity' is not applicable with other arguments")
+    if any([password, service_principal, username, identity]) and use_device_code:
+        raise CLIError("usage error: '--use-device-code' is not applicable with other arguments")
```

Comment:
Why do we need a new parameter for this?  If you log in interactively why wouldn't you always want to use the device code?

Reply:
it is just for safeness that in case people still like to use the old device code.